### DISCUSSION TITLE
Error during the build process

### DISCUSSION THREAD

**Discussion by bssudhir** [awslabs/aws-saas-boost#58] [Q&A] [answered]
Hi,

I installed all the pre-requisites as per the documentation. However, when I run the install.sh after getting the code, I get the following error. I could not find any logs either. FYI... I have my build VM behind a proxy.

Start SaaS Boost Installation
AWS Region = <not
Build installer jar with maven
Error with build of Java installer for SaaS Boost

**Comment by brtrvn** [comment]
Take a look in the `saas-boost-install.log` file for more details on what the error is.

**Reply by bssudhir** [reply]
Appreciate the quick response. Thank you. 

Unfortunately, no error logs are being generated. I searched for the log file on my entire VM. My VM is running RHEL 8.2. Not sure that is going to make any difference.

Regards,
Sudhir

**Comment by 4patelr** [comment]
Did you try setting the AWS_REGION environment variable before running install.sh?

**Reply by bssudhir** [reply]
Hi, 

I was able to set the region and required keys using the aws cli command as per the instructions in https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html

Once I fixed it, I was able to proceed further but ran into the below issue. I tried with both 14.15.1 and 14.17 versions of NodeJS and I am getting the same error. This is because when I run the version command, I am getting "10.24.0" but the script expects a version higher than 12. Please see the output below.

[root@sudhir_saas_boost_vm bin]# pwd
**/root/node-v14.17.0-linux-x64/bin**
[root@sudhir_saas_boost_vm bin]# ll
total 72516
-rwxr-xr-x. 1 scs scs 74253640 May 11 11:14 node
lrwxrwxrwx. 1 scs scs       38 May 11 11:14 npm -> ../lib/node_modules/npm/bin/npm-cli.js
lrwxrwxrwx. 1 scs scs       38 May 11 11:14 npx -> ../lib/node_modules/npm/bin/npx-cli.js
[root@sudhir_saas_boost_vm bin]# node -v
**v10.24.0**


_Start SaaS Boost Installation
AWS Region = us-west-2
Build installer jar with maven
Download dependencies for React Web App
yarn install v1.22.10
info No lockfile found.
[1/5] Validating package.json...
**error sass-boost-web@0.1.0: The engine "node" is incompatible with this module. Expected version ">=12". Got "10.24.0"**
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error with yarn build of dependencies for React Web App. Check node version per documentation_

**Reply by 4patelr** [reply]
Did you set up your environment to use the Node 14.x version of Node that you installed? You can use "nvm" to manage node versions and run "nvm use 14" to use version 14 before you run the install.sh. nvm can be found at https://github.com/nvm-sh/nvm.

**Reply by bssudhir** [reply]
Thank you very much for the help.

After installing nvm, I was able to install the 14.15.0 version of nodejs and continue with the build process. Finally, the build was successful but during the installation, I am seeing the below error. I am also attaching the install log for your reference. Can this issue be because that I have my build VM behind a firewall?
[saas-boost-install.log](https://github.com/awslabs/aws-saas-boost/files/6605759/saas-boost-install.log)

_Launch Java Installer for SaaS Boost
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
Setting version to v0 as it is missing from the git properties file.

Welcome to the AWS SaaS Boost Installer
Setting version to v0 as it is missing from the git properties file.
Installer Version: b6b1704, Commit time: 2021-06-01T18:42:42-0700
Checking maven, yarn and AWS CLI...
Environment Checks for maven, yarn, and AWS CLI PASSED.

Installation Error: Unable to execute HTTP request: connect timed out
Please see detailed log file saas-boost-install.log
software.amazon.awssdk.core.exception.SdkClientException: Unable to execute HTTP request: connect timed out
        at software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98)
        at software.amazon.awssdk.core.exception.SdkClientException.create(SdkClientException.java:43)_

**Comment by brtrvn** [comment, marked as answer]
Sudhir - your guess about network connectivity may be correct. The error you received was a timeout trying to connect to the IAM service endpoint. Please see https://docs.aws.amazon.com/general/latest/gr/iam-service.html for a list of endpoints by region. See if you can `ping` and/or `traceroute` the endpoint from the VM you're working on.

**Reply by bssudhir** [reply]
Thanks for the confirmation. I brought up a build VM in AWS and was able to complete the installation successfully. Appreciate all the help.
